woocommerce/woocommerce-android
# Issue/4878 Location fetching error handling

Closes: #4878 

### Description

The PR adds handling of the errors for location fetching endpoint. When a merchant doesn't have an address set in WC Pay, then the backend returns a specific error. We show a screen with an explanation and a button that opens a web view on the settings page where a merchant can set an address.

Other errors were handled in a generic way - we should that we couldn't connect to a reader

### Testing instructions


** Your site should have the current develop (3.1.1) version of WCPay install, please contact me if you don't know where to download that**

** Missing address**
1. Remove WC pay address via wp-admin
2. Adjust `CardReaderConnectViewModel::connectToReader:332` to `val cardReaderLocationId = if (cardReader.locationId != null) null else cardReader.locationId` (this way we always featch a location id from the backend)
3. Try to connect to a reader
4. Notice specific dialog 
5. Notice that web view opens and you can change the address
6. Try to fill out the address
7. Try to connect to a reader
8. Notice that you can connect now

**Generic error**
1. Do 2 from the previous list
2. Try to connect to a reader a disable internet
3. Notice error dialog

### Images/gif


https://user-images.githubusercontent.com/4923871/137124006-10f938ba-d587-47a8-97a6-dd470ef9680c.mp4
